drivers/flash/soc_flash_mcux.c

```diff
@@ -14,7 +14,11 @@
 #include "flash_priv.h"
 
 #include "fsl_common.h"
+#ifdef CONFIG_SOC_FAMILY_LPC
```

Comment:
It would be more consistent with the Kconfig dependency to use `CONFIG_HAS_MCUX_IAP` here